kratos/tests/cpp_tests/elements/test_levelset_convection_element_simplex.cpp

```diff
@@ -0,0 +1,277 @@
+        p_conv_diff_settings->SetUnknownVariable(DISTANCE);
+        p_conv_diff_settings->SetConvectionVariable(VELOCITY);
+
+        // Set the element properties
+        Properties::Pointer pElemProp = model_part.CreateNewProperties(0);
+
+        // Geometry creation
+        model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+        model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+        model_part.CreateNewNode(3, 0.0, 1.0, 0.0);
+        std::vector<ModelPart::IndexType> elemNodes {1, 2, 3};
+        auto p_element = model_part.CreateNewElement("LevelSetConvectionElementSimplex2D3N", 1, elemNodes, pElemProp);
+        
+        // Define the nodal values
+        Matrix vel_original(3,2);
+        vel_original(0,0) = 0.0; vel_original(0,1) = 0.1;
+        vel_original(1,0) = 0.1; vel_original(1,1) = 0.2;
+        vel_original(2,0) = 0.2; vel_original(2,1) = 0.3;
+
+        for(unsigned int i=0; i<3; i++){
+            for(unsigned int k=0; k<2; k++){
+                p_element->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY)[k]    = vel_original(i,k);
+                p_element->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY, 1)[k] = 0.9*vel_original(i,k);
+                p_element->GetGeometry()[i].FastGetSolutionStepValue(VELOCITY, 2)[k] = 0.75*vel_original(i,k);
+            }
+        }
+        p_element->GetGeometry()[0].FastGetSolutionStepValue(DISTANCE) = -1.0;
+        p_element->GetGeometry()[1].FastGetSolutionStepValue(DISTANCE) = -1.0;
+        p_element->GetGeometry()[2].FastGetSolutionStepValue(DISTANCE) =  0.5;
+
+        p_element->GetGeometry()[0].FastGetSolutionStepValue(DISTANCE,1) = -0.8;
+        p_element->GetGeometry()[1].FastGetSolutionStepValue(DISTANCE,1) = -0.8;
+        p_element->GetGeometry()[2].FastGetSolutionStepValue(DISTANCE,1) =  0.7;
+
+        // Compute RHS and LHS
+        Vector rhs = ZeroVector(4);
+        Matrix lhs = ZeroMatrix(4,4);
+
+        p_element->Initialize();
+        const ProcessInfo& const_process_info = model_part.GetProcessInfo();
+        p_element->CalculateLocalSystem(lhs, rhs, const_process_info);
+
+        KRATOS_CHECK_EQUAL(rhs.size(), 3);
+        KRATOS_CHECK_EQUAL(lhs.size1(), 3);
+        KRATOS_CHECK_EQUAL(lhs.size2(), 3);
+        KRATOS_CHECK_NEAR(rhs(0), 0.1186006876 , 1e-7);
+        KRATOS_CHECK_NEAR(rhs(1), 0.4123679368 , 1e-7);
+        KRATOS_CHECK_NEAR(rhs(2), 0.3265313756 , 1e-7);
+        KRATOS_CHECK_NEAR(lhs(0,0), 0.367583692212 , 1e-7);
+        KRATOS_CHECK_NEAR(lhs(0,1), -0.301736964939 , 1e-7);
+        KRATOS_CHECK_NEAR(lhs(0,2), -0.336905789443 , 1e-7);
+        KRATOS_CHECK_NEAR(lhs(1,0), 0.493383924505 , 1e-7);
+        KRATOS_CHECK_NEAR(lhs(1,1), 1.1281300063 , 1e-7);
+        KRATOS_CHECK_NEAR(lhs(1,2), 0.677825753415 , 1e-7);
+        KRATOS_CHECK_NEAR(lhs(2,0), 0.73444904995 , 1e-7);
+        KRATOS_CHECK_NEAR(lhs(2,1), 0.864023625302 , 1e-7);
+        KRATOS_CHECK_NEAR(lhs(2,2), 1.37324670269 , 1e-7);        
```

Comment:
This increases a lot the compilation time. It is much more efficient to set two arrays (one with the expected and one with the obtained values) and loop them just calling the macro once.

Reply:
Forget about this (see @philbucher comment below and use the array comparison macros)